src/use-dropdown-menu.ts

```diff
@@ -12,10 +12,11 @@ export interface ButtonProps<ButtonElement extends HTMLElement>
 
 // Create interface for item properties
 export interface ItemProps {
-	onKeyDown: (e: React.KeyboardEvent<HTMLAnchorElement>) => void;
+	onKeyDown: (e: React.KeyboardEvent<HTMLElement>) => void;
 	tabIndex: number;
 	role: string;
-	ref: React.RefObject<HTMLAnchorElement>;
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	ref: React.RefObject<any>;
```

Comment:
We try to avoid using `any` within our codebases since it typically causes a loss of type safety and can more easily lead to bugs. That's why we use the `@typescript-eslint/no-explicit-any` ESLint rule. Can you look into designing this without `any`?

Reply:
I did and as I said: „We can make it a generic parameter, but it wouldn't bring any real benefit, just increase complexity and decrease flexibility.“ Using `any` is perfectly fine in this case, it does not cause a loss of type safety for the users of this hook.

Reply:
I read what you wrote in your PR description; however, what you've stated is not true.

To demonstrate this, let's say a user of this hook needs to write a `useEffect()` hook with one of the item's `ref` objects. They might add this code to the `src/use-dropdown-menu.test.tsx` file:

```tsx
useEffect(() => {
	// Do something
}, [itemProps[0].ref.current?.nonexistentProperty]);
```

Currently, this correctly throws a compile error:

<img width="387" alt="Screenshot 2024-10-17 at 8 30 05 AM" src="https://github.com/user-attachments/assets/b2304e06-6d20-4b76-9497-b47377d0d818">

The same `useEffect()` hook with your proposed changes gives no warning:

<img width="385" alt="Screenshot 2024-10-17 at 8 31 22 AM" src="https://github.com/user-attachments/assets/23a4c1ac-48eb-4ad5-91fa-8201236280fa">

So using `any` in this case would cause a loss of type safety whenever a user of this package needs to work with a `ref` on one of the items.